receiver/statsdreceiver/transport/client/client.go

```diff
@@ -42,33 +34,22 @@ func NewStatsD(transport Transport, host string, port int) (*StatsD, error) {
 }
 
 // connect populates the StatsD.Conn
-func (s *StatsD) connect(transport Transport) error {
-	if cl, ok := s.Conn.(io.Closer); ok {
```

Comment:
Just a curiosity, what were they doing here?

Reply:
To be honest, I don't know. while testing, the client was closing the connection before start sending any data. It makes no sense and it was giving issues with the socket as it started to close it.